packages/common/pipes/file/parse-file.pipe.ts

```diff
@@ -48,7 +48,11 @@ export class ParseFilePipe implements PipeTransform<any> {
     }
 
     if (this.validators.length) {
-      await this.validate(value);
+      if (Array.isArray(value)) {
+        await Promise.all(value.map((v) => this.validate(v)));
+      } else {
+        await this.validate(value);
+      }
```

Comment:
This is a nit as well, but we have an `if` statement nested here along with the `else` condition, increasing a bit of the cognitive complexity here. Would you mind also wrapping this block inside a private method? Something like:

```suggestion
      this.validateFilesFrom(value)
```

and inside this private method, you just paste the same logic.